framework/src/main/java/org/tron/core/Wallet.java

```diff
@@ -4127,12 +4129,11 @@ public byte[] getShieldedContractScalingFactor(byte[] contractAddress)
     TransactionExtention.Builder trxExtBuilder = TransactionExtention.newBuilder();
     Return.Builder retBuilder = Return.newBuilder();
     TransactionExtention trxExt;
-    Transaction trx;
 
     try {
       TransactionCapsule trxCap = createTransactionCapsule(trigger,
           ContractType.TriggerSmartContract);
-      trx = triggerConstantContract(trigger, trxCap, trxExtBuilder, retBuilder);
+      Transaction trx = triggerConstantContract(trigger, trxCap, trxExtBuilder, retBuilder);
```

Comment:
Doesn't this need to be handled?

Reply:
This part is rollback to 4.7.3.

Reply:
What is the reason for this part of the direct rollback ? I see the previous analysis which is also wrong

Reply:
Only this part is added: 
```
if (scalingFactor.compareTo(BigInteger.ZERO) <= 0) {
      throw new ContractValidateException("scalingFactor must be positive");
}
```

Other part is still same as before. Check the condition outside this method is not only easy to understand and but also include all conditions.

Reply:
@halibobo1205 what's wrong with previous analysis? This optimization is to coverage this additional case: contract and method `scalingFactor` are both exist, but it really return 0.